src/bin/ion/commands/beta/generate/generator.rs

```diff
@@ -0,0 +1,466 @@
+use crate::commands::beta::generate::context::{CodeGenContext, DataModel};
+use crate::commands::beta::generate::result::{invalid_data_model_error, CodeGenResult};
+use crate::commands::beta::generate::utils::{Field, Import, Language};
+use convert_case::{Case, Casing};
+use ion_schema::isl::isl_constraint::{IslConstraint, IslConstraintValue};
+use ion_schema::isl::isl_type::IslType;
+use ion_schema::isl::isl_type_reference::IslTypeRef;
+use ion_schema::isl::IslSchema;
+use std::collections::HashMap;
+use std::fs;
+use std::fs::File;
+use std::io::Write;
+use std::path::Path;
+use tera::Context;
+
+pub(crate) struct CodeGenerator;
+
+impl CodeGenerator {
+    /// Maps the given type name to a native type based on programming language
+    pub fn map_to_base_type(name: &str, language: &Language) -> String {
```

Comment:
Consider making this a method on `Language` instead. When we get to supporting 4+ target languages, it'll be easier to see how each language handles its mappings individually.